### PR TITLE
Optimise -Improved developer interface 

### LINKED ISSUE #6: Third evaluation
- [x] Finish the main module, with its developer interface and complete interaction between the two modules
- [ ] Write unit tests for The main Module and also write the library’s readme file for documentation.
- [x] GSOC blog post

### LINKED ISSUE #15: Textarea inputs
The editor is missing support for textarea input types. This kind of HTML input is necessary for fields like the contents of files.

### LINKED ISSUE #16: Examples: set examples/index.html as first page of dev server
Could we set [examples/index.html](https://github.com/openwisp/netjsonconfig-editor.js/blob/master/examples/index.html) as the first page of the dev server (`npm run dev`)?

Also, what is [examples/index1.html](https://github.com/openwisp/netjsonconfig-editor.js/blob/master/examples/index1.html) about? I can't see anything useful when I try access it from the development server. If it's not useful anymore could you remove it?

### LINKED ISSUE #17: Use git repo as a dependency instead of git submodule
@gastonche can you do something like this instead of the git submodule?
https://docs.npmjs.com/files/package.json#git-urls-as-dependencies

It would be a lot simpler and would save us a lot of troubles.

### PR DESCRIPTION
**Pull request by gastonche**
Changes in this PR:
 - Added events setters `on` and clearer `off`
- Improved the` readme` file
- Added more functions to improve developer interactions with the editor
- some UI changes
- upated some packages

Closes #15
Closes #17 
Closes #16 
Closes #6

### CODE REVIEW

**Comment by nemesisdesign** [comment]
What about custom properties? I tried adding a custom property and it doesn't show up in the UI :-(

**Comment by gastonche** [comment]
I am currently trying to implement the feature for extra fields on the liform-react library in order to enable custom fields. How ever I have not finished it yet. I have submitted another PR which is yet to be merged. I'll push that after the current PR at [liform-react](https://github.com/Limenius/liform-react/pull/13)

**Comment by gastonche** [comment]
To be able to test the latest updates to this PR you will need to install some new packages,
run: `npm install lodash deepmerge classnames`

File: examples/index.html [hunk 1]
```diff
@@ -3558,16 +3558,145 @@
...
+        {
+            "config_value": "vlan31",
+            "config_name": "Interface",
+            "interface": "vlan31"
+        }
+    ]
+}, 
 			onChange: function(data){
-                console.log(data);
+                // console.log(data);
```
Comment (nemesisdesign) [inline examples/index.html]: remove this

**Comment by gastonche** [comment]
just set the dev server to load into the examples dir when `npm run dev` is run. Fixing #16

**Comment by gastonche** [comment]
Added features to solve the following issues #15 #17

**Pull request merged**